src/vs/workbench/services/positronDataExplorer/common/tableDataCache.ts

```diff
@@ -0,0 +1,858 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2024 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Emitter } from 'vs/base/common/event';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { arrayFromIndexRange } from 'vs/workbench/services/positronDataExplorer/common/utils';
+import { DataExplorerClientInstance } from 'vs/workbench/services/languageRuntime/common/languageRuntimeDataExplorerClient';
+import { ArraySelection, ColumnSchema, ColumnSelection, DataSelectionIndices, DataSelectionRange } from 'vs/workbench/services/languageRuntime/common/positronDataExplorerComm';
+
+/**
+ * Constants.
+ */
+const TRIM_CACHE_TIMEOUT = 3000;
+const OVERSCAN_FACTOR = 3;
+const CHUNK_SIZE = 4096;
+
+/**
+ * InvalidateCacheFlags enum.
+ */
+export enum InvalidateCacheFlags {
+	None = 0,
+	ColumnSchema = 1 << 0,
+	Data = 1 << 1,
+	All = ~(~0 << 2)
```

Comment:
Would a more readable way of specifying an "All" flag enum be:
```
All = ColumnSchema | Data
```

Reply:
Too clever by half. I've done as you suggested.